src/main/java/org/unicitylabs/sdk/verification/VerificationRule.java

```diff
@@ -2,20 +2,37 @@
 
 import java.util.Objects;
 
-public abstract class VerificationRule<CTX> {
+/**
+ * Verification rule base class.
+ *
+ * @param <ContextT> verification context
+ */
+public abstract class VerificationRule<ContextT> {
```

Comment:
`ContextT` looks vague. is it possible to add the base class to make it more specific? If UnicityCertificateVerificationContext is the only implementation, maybe just use it directly?